src/main/java/org/truffleruby/parser/ast/ArgsParseNode.java

```diff
@@ -127,9 +128,17 @@ private Arity createArity() {
                 final AssignableParseNode assignableNode = kwarg.getAssignable();
 
                 if (assignableNode instanceof LocalAsgnParseNode) {
-                    keywordArguments[i] = ((LocalAsgnParseNode) assignableNode).getName();
+                    final LocalAsgnParseNode node = ((LocalAsgnParseNode) assignableNode);
+                    keywordArguments[i] = node.getName();
+                    if (isAllKeywordsOptional) {
+                        isAllKeywordsOptional = !(node.getValueNode() instanceof RequiredKeywordArgumentValueParseNode);
+                    }
                 } else if (assignableNode instanceof DAsgnParseNode) {
-                    keywordArguments[i] = ((DAsgnParseNode) assignableNode).getName();
+                    final DAsgnParseNode node = ((DAsgnParseNode) assignableNode);
+                    keywordArguments[i] = node.getName();
```

Comment:
Not sure if we need to DRY these two branches

Reply:
We should be able because there is `AssignableParseNode#getValueNode()`:
```java
isAllKeywordsOptional &= !(assignableNode.getValueNode() instanceof RequiredKeywordArgumentValueParseNode);
```
or to reuse the helper to be more concise and avoid duplication:
```java
isAllKeywordsOptional &= !Helpers.isRequiredKeywordArgumentValueNode(assignableNode);
```

Reply:
Look like you are familiar with all existing code :)

I like using `Helpers` approach - done  :+1: